src/insert/spacefinder/spacefinder.ts

```diff
@@ -198,22 +204,85 @@ const partitionCandidates = <T>(
+ *
+ *                                                        │
+ *                     Opponent Below                     │             Opponent Above
+ *                                                        │
+ *                  ───────────────────  Top of container │          ───────────────────  Top of container
+ *                    ▲              ▲                    │            ▲              ▲
+ *                    │              │                    │            │              │ opponent.top
+ *                    │ ┌──────────┐ │                    │            │ ┌──────────┐ ▼   (insertion point)
+ *                    │ │          │ │opponent.bottom     │            │ │          │
+ *                    │ │ Candidate│ │                    │            │ │ Opponent |
+ *       opponent.top │ │          │ │                    candidate.top│ │          │
+ *                    │ └──────────┘ ▼                    │            │ └──────────┘
+ *                    │           ▲                       │            │
+ *                    │           │ minBelow              │            │  ───────────
+ *                    │           ▼                       │            │           ▲
+ *                    │ ────────────                      │            │           │ minAbove
+ *                    │                                   │            │           ▼
+ * (insertion point)  ▼ ┌──────────┐                      │            ▼ ┌──────────┐
+ *                      │          │                      │              │          │
+ *                      │ Opponent |                      │              │ Candidate│
+ *                      │          │                      │              │          │
+ *                      └──────────┘                      │              └──────────┘
+ *                                                        │
+ *                                                        │
+ */
+const isTopOfCandidateFarEnoughFromOpponent = (
+	candidate: SpacefinderItem,
+	opponent: SpacefinderItem,
+	rule: RuleSpacing,
+	isOpponentBelow: boolean,
+): boolean => {
+	const potentialInsertPosition = candidate.top;
+
+	if (isOpponentBelow && rule.minBelowSlot) {
+		return opponent.top - potentialInsertPosition >= rule.minBelowSlot;
+	}
+
+	if (!isOpponentBelow && rule.minAboveSlot) {
+		return potentialInsertPosition - opponent.bottom >= rule.minAboveSlot;
+	}
+
+	// if no rule is set (or they're 0), return true
+	return true;
+};
+
 // test one element vs another for the given rules
 const testCandidate = (
 	rule: RuleSpacing,
 	candidate: SpacefinderItem,
 	opponent: SpacefinderItem,
 ): boolean => {
-	const isMinAbove = candidate.top - opponent.bottom >= rule.minAbove;
-	const isMinBelow = opponent.top - candidate.top >= rule.minBelow;
+	if (candidate.element === opponent.element) {
```

Comment:
Now `h2`'s can be candidates but we also want to avoid placing ads under them, without this check it will try to avoid itself!

Reply:
Why did paragraph's not need to have this check? Is it because a paragraph is never an opponent?

Reply:
Yes previously candidate selector and opponent selectors never overlapped, but with h2's now they do